### PR TITLE
Add async validation block.

### PR DESCRIPTION
**Pull request by wzs**
Hi,

one thing I missed with this component is asynchronous validation - i.e. perform some backend validation.
This pull presents simple approach to this problem.

### CODE REVIEW

File: Example/RootViewController.m [hunk 1]
```diff
@@ -30,6 +30,32 @@ - (void)viewDidLoad
...
+            return NO;
+        }
+
+        BOOL isValid = [json[@"is_valid"] boolValue];
+        if (!isValid) {
+            weakSelf.emailField.alertView.title = @"Invalid email address (online)";
+        }
+                
+        return isValid;
+    }];
```
Comment (0thernet) [inline Example/RootViewController.m:57]: awesome! shameless plug: https://github.com/benzguo/BZGMailgunEmailValidation :wink:

File: BZGFormField/BZGFormField.m [hunk 2]
```diff
@@ -113,10 +121,51 @@ - (void)setup
...
+    CABasicAnimation *animation = [CABasicAnimation animationWithKeyPath:@"backgroundColor"];
+    animation.fromValue = (id)self.leftIndicatorValidColor.CGColor;
+    animation.toValue = (id)self.leftIndicatorNoneColor.CGColor;
+    animation.autoreverses = YES;
+    animation.repeatCount = HUGE_VALF;
+    animation.duration = 0.75f;
+    animation.timingFunction = [CAMediaTimingFunction functionWithName:kCAMediaTimingFunctionEaseInEaseOut];
+    
+    return animation;
+}
```
Comment (0thernet) [inline BZGFormField/BZGFormField.m:140]: nice!

**Comment by 0thernet** [comment]
:octocat: = :smile_cat:

**Comment by wzs** [comment]
Thanks - good that I havn't checked your's BZGMailgunEmailValidation first :)

**Pull request merged**